hrm-domain/hrm-core/jobs/search/publishToSearchIndex.ts

```diff
@@ -46,14 +76,89 @@ const publishToSearchIndex = async ({
   }
 };
 
+const publishToSns = async ({
+  entityType,
+  payload,
+  messageGroupId,
+}: {
+  entityType: 'contact' | 'case';
+  payload: NotificationPayload;
+  messageGroupId: string;
+}) => {
+  try {
+    const topicArn = await getSsmParameter(getSnsSsmPath(entityType));
+    const publishParameters: PublishSnsParams = {
+      topicArn,
+      message: JSON.stringify({ ...payload, entityType }),
+      messageGroupId,
+      messageAttributes: { operation: payload.operation },
+    };
+    console.debug('Publishing HRM entity update:', publishParameters);
+    return await publishSns(publishParameters);
+  } catch (err) {
+    if (err instanceof SsmParameterNotFound) {
+      console.debug(
+        `No SNS topic stored in SSM parameter ${getSnsSsmPath(
+          entityType,
+        )}. Skipping publish.`,
+      );
+      return;
+    }
+    console.error(
+      `Error trying to publish message to SNS topic stored in SSM parameter ${getSnsSsmPath(
+        entityType,
+      )}`,
+      err,
+    );
+  }
+};
+
+const publishEntityToSearchIndex = async (
+  accountSid: HrmAccountId,
+  entityType: 'contact' | 'case',
+  entity: Contact | CaseService,
+  operation: IndexMessage['operation'] | 'reindex',
+) => {
+  const messageGroupId = `${accountSid}-${entityType}-${entity.id}`;
+  if (operation === 'remove') {
+    await publishToSns({
+      entityType,
+      payload: { accountSid, id: entity.id.toString(), operation: 'delete' },
+      messageGroupId,
+    });
+  } else {
+    const publishOperation = operation === 'reindex' ? 'reindex' : 'update';
+    await publishToSns({
+      entityType,
+      // Update / create are identical for now, will differentiate between the 2 ops in a follow pu refactor PR
+      payload: {
+        accountSid,
+        [entityType]: entity,
+        operation: publishOperation,
+      } as NotificationPayload,
+      messageGroupId,
+    });
+  }
+  const indexOperation = operation === 'reindex' ? 'index' : operation;
+  return publishToSearchIndex({
+    message: {
+      accountSid,
+      type: entityType,
+      [entityType]: entity,
+      operation: indexOperation,
+    } as IndexMessage,
+    messageGroupId: `${accountSid}-${entityType}-${entity.id}`,
+  });
```

Comment:
I thought we will only send the message to the SNS topic, and it will send to the appropriate queue accordingly - e.g. we can add "targets" to each operation and use them as SNS filters? Anyways, I'm not suggesting we change this now, is just that it seems unnecessary to have the SNS topic if we'll send to the queues separately anyways.

Reply:
At the moment the the search index queue isn't subscribing to the SNS topic, only the DW queue it. I just wanted to get the end to end flow working for the DW working first, before migrating systems that are already live in prod. We can migrate the search index queue to ging via the SNS topic anon

One concern I had about migrating straight away was that SNS puts an extra wrapper on the messages sent to SQS that aren't there on messages sent directly, but I've found out there is a setting that should prevent it doing that so the migration _should_ be straightforward